frontend/storm_wiki/util/storm_runner.py

```diff
@@ -0,0 +1,472 @@
+                stop=("\n\n---",),
+            )
+        elif model_type == "openai":
+            logger.info(
+                f"Creating OpenAI client with model: {model_settings['openai']['model']}"
+            )
+            return OpenAIModel(
+                model=model_settings["openai"]["model"],
+                api_key=os.getenv("OPENAI_API_KEY"),
+                max_tokens=model_settings["openai"]["max_tokens"],
+            )
+        elif model_type == "anthropic":
+            logger.info(
+                f"Creating Anthropic client with model: {model_settings['anthropic']['model']}"
+            )
+            client = ClaudeModel(
+                model=model_settings["anthropic"]["model"],
+                api_key=os.getenv("ANTHROPIC_API_KEY"),
+                max_tokens=model_settings["anthropic"]["max_tokens"],
+                temperature=1.0,
+                top_p=0.9,
+            )
+            logger.info("Anthropic client created successfully")
+            return client
+        else:
+            raise ValueError(f"Unsupported model type: {model_type}")
+    except Exception as e:
+        logger.error(f"Error creating {model_type} client: {str(e)}", exc_info=True)
+        if fallback and fallback_model:
+            logger.warning(f"Falling back to {fallback_model}")
+            return create_lm_client(
+                fallback_model, fallback=False, model_settings=model_settings
+            )
+        else:
+            raise
+
+
+def run_storm_with_config(
+    topic: str,
+    current_working_dir: str,
+    callback_handler=None,
+):
+    progress_placeholder = st.empty()
+
+    def update_progress(message):
+        progress_placeholder.write(message)
+        if callback_handler:
+            callback_handler.on_information_gathering_start(message=message)
+        logger.info(message)
+
+    update_progress("Loading configurations...")
+    llm_settings = load_llm_settings()
+    search_options = load_search_options()
+    search_top_k = search_options["search_top_k"]
+    retrieve_top_k = search_options["retrieve_top_k"]
+
+    update_progress("Setting up LLM...")
+    primary_model = llm_settings["primary_model"]
+    fallback_model = llm_settings["fallback_model"]
+    model_settings = llm_settings["model_settings"]
+    logger.info(f"Primary model: {primary_model}, Fallback model: {fallback_model}")
+    logger.info(f"Model settings: {json.dumps(model_settings, indent=2)}")
+
+    llm_configs = STORMWikiLMConfigs()
+
+    try:
+        primary_lm = create_lm_client(
+            primary_model,
+            fallback=False,
+            model_settings=model_settings,
+        )
+
+        fallback_lm = None
+        if fallback_model:
+            fallback_lm = create_lm_client(
+                fallback_model,
+                fallback=False,
+                model_settings=model_settings,
+            )
+
+        for lm_type in [
```

Comment:
I think we may want to allow people to configure LMs for these components separately. Currently, we can only use the same LM for all these components based on the interface design. One design philosophy of STORM is that it's a LM-empowered system so different modules can be run with different LMs to jointly optimize the cost and quality. I think viewing the system in this way will be better than viewing it as a chain of GPT-4/a specific LM prompts.

No need to adjust it this in this PR though.